libr/core/cmd_seek.c

```diff
@@ -138,341 +138,401 @@ R_API int r_core_lines_initcache (RCore *core, ut64 start_addr, ut64 end_addr) {
-			}
+			free (cb.str);
+		} else {
+			const char *help_msg[] = {
+				"Usage:", "sC", "Comment grep",
+				"sC", "*", "List all comments",
+				"sC", " str", "Seek to the first comment matching 'str'",
+				NULL
+			};
+			r_core_cmd_help (core, help_msg);
+		}
+		break;
+	case ' ':
+		r_io_sundo_push (core->io, core->offset, r_print_get_cursor (core->print));
+		r_core_seek (core, off * sign, 1);
+		r_core_block_read (core);
+		break;
+	case '/':
+	{
+		const char *pfx = r_config_get (core->config, "search.prefix");
+		ut64 from = r_config_get_i (core->config, "search.from");
+// kwidx cfg var is ignored
+		int kwidx = core->search->n_kws; // (int)r_config_get_i (core->config, "search.kwidx")-1;
+		if (kwidx < 0) {
+			kwidx = 0;
+		}
+		switch (input[1]) {
+		case ' ':
+		case 'v':
+		case 'V':
+		case 'w':
+		case 'W':
+		case 'z':
+		case 'm':
+		case 'c':
+		case 'A':
+		case 'e':
+		case 'E':
+		case 'i':
+		case 'R':
+		case 'r':
+		case '/':
+		case 'x':
+			r_config_set_i (core->config, "search.from", core->offset + 1);
+			r_config_set_i (core->config, "search.count", 1);
+			r_core_cmdf (core, "s+1; %s; s-1; s %s%d_0; f-%s%d_0",
+				input, pfx, kwidx, pfx, kwidx, pfx, kwidx);
+			r_config_set_i (core->config, "search.from", from);
+			r_config_set_i (core->config, "search.count", 0);
 			break;
-		case '-':
-			if (input[1]!='\0') {
-				int delta = (input[1]=='-') ? -core->blocksize: -off;
-				r_io_sundo_push (core->io, core->offset, r_print_get_cursor (core->print));
-				r_core_seek_delta (core, delta);
-			} else {
-				RIOUndos *undo = r_io_sundo (core->io, core->offset);
-				if (undo) {
-					r_core_seek (core, undo->off, 0);
-					r_core_block_read (core);
-				}
-			}
+		case '?':
+			eprintf ("Usage: s/.. arg.\n");
+			r_cons_printf ("/?\n");
 			break;
-		case 'n':
-			r_io_sundo_push (core->io, core->offset, r_print_get_cursor (core->print));
-			r_core_seek_next (core, r_config_get (core->config, "scr.nkey"));
+		default:
+			eprintf ("unknown search method\n");
 			break;
-		case 'p':
+		}
+	}
+	break;
+	case '.':
+		for (input++; *input == '.'; input++) {
+			;
+		}
+		r_core_seek_base (core, input);
+		break;
+	case '*':
+	case '=':
+	case 'j':
+		r_io_sundo_list (core->io, input[0]);
+		break;
+	case '+':
+		if (input[1] != '\0') {
+			int delta = (input[1] == '+')? core->blocksize: off;
 			r_io_sundo_push (core->io, core->offset, r_print_get_cursor (core->print));
-			r_core_seek_previous (core, r_config_get (core->config, "scr.nkey"));
-			break;
-		case 'a':
-			off = core->blocksize;
-			if (input[1]&&input[2]) {
-				cmd = strdup (input);
-				p = strchr (cmd+2, ' ');
-				if (p) {
-					off = r_num_math (core->num, p+1);;
-					*p = '\0';
-				}
-				cmd[0] = 's';
-				// perform real seek if provided
-				r_cmd_call (core->rcmd, cmd);
-				free (cmd);
+			r_core_seek_delta (core, delta);
+		} else {
+			RIOUndos *undo = r_io_sundo_redo (core->io);
+			if (undo != NULL) {
```

Comment:
Ok. But please notice that the first commit (that includes most of the "changes" you see here it's just indentation. I didn't change/review the code.